### DISCUSSION TITLE
Understanding why fastify-mongodb needs to be used

### DISCUSSION THREAD

**Discussion by xzilja** [fastify/fastify#3906] [Q&A] [answered]
Hey everyone 👋 

I'm building out an api using fastify and mongodb and saw that you provide a plugin called `fastify-mongodb`. I'm trying to understand why it is needed as opposed to using normal mongodb js sdk? I understand that under the hood, this is what plugin uses as well.

Does this bring any performance benefits or it is mainly for convenience?

**Comment by mcollina** [comment, marked as answer]
convenience. You want to avoid creating a global connection to mongodb as that will make testing more complicated.
